R/use_crate.R

```diff
@@ -0,0 +1,141 @@
+#' Add dependencies to a Cargo.toml manifest file
+#'
+#' Analogous to `usethis::use_package()` but for crate dependencies.
+#'
+#' @param crate a character scalar, the name of the crate to add
+#' @param features a character vector, a list of features to include from the
+#' crate
+#' @param git a character scalar, the URL of the Github repository
+#' @param version a character scalar, the version of the crate to add
+#' @param path a character scalar, the package directory
+#' @param ... additional options to include
+#'
+#' @details
+#' For a list of all available options, see \href{https://doc.rust-lang.org/cargo/commands/cargo-add.html}{Cargo docs}
+#' for `cargo-add`.
+#'
+#' @return `NULL`, invisibly
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # add to [dependencies]
+#' use_crate("serde")
+#'
+#' # add to [dependencies] and [features]
+#' use_crate("serde", features = "derive")
+#'
+#' # add to [dependencies] using github repository as source
+#' use_crate("serde", git = "serde-rs/serde")
+#'
+#' # add to [dependencies] with specific version
+#' use_crate("serde", version = "1.0.1")
+#' }
+use_crate <- function(
+    crate,
+    features = NULL,
+    git = NULL,
+    version = NULL,
+    path = ".",
+    ...
+){
+
+  # check crate
+  if (!rlang::is_character(crate) && length(crate) != 1){
+
+    cli::cli_abort(
+      "{.var crate} should be a length one character vector.",
+      "i" = "You supplied an object of class {class(crate)} with length {length(crate)}.",
+      class = "rextendr_error"
+    )
+
+  }
+
+  # check features
+  if (!is.null(features) && !rlang::is_character(features)){
+
+    cli::cli_abort(
+      "{.var features} should be a character vector.",
+      "i" = "You supplied an object of class {class(features)}.",
```

Comment:
this `class()` call should grab the first element of the output. There may be 1 or more classes

Reply:
good point. are you thinking `class(x)[1]` or is there a better way?

Reply:
that works for me! We could also utilize the `obj_type_friendly()` standalone function.
https://github.com/r-lib/rlang/blob/main/R/standalone-obj-type.R